src/Datasource/EntityTrait.php

```diff
@@ -375,6 +375,87 @@ public function has($property)
     }
 
     /**
+     * Returns whether this entity contains a property named $property
+     * that contains a non-null value.
+     *
+     * ### Example:
+     *
+     * ```
+     * $entity = new Entity(['id' => 1, 'name' => null]);
+     * $entity->has('id'); // true
+     * $entity->has('name'); // false
+     * $entity->has('last_name'); // false
+     * ```
+     *
+     * You can check multiple properties by passing an array:
+     *
+     * ```
+     * $entity->has(['name', 'last_name']);
+     * ```
+     *
+     * All properties must not be null to get a truthy result.
+     *
+     * When checking multiple properties. All properties must not be null
+     * in order for true to be returned.
+     *
+     * @param string|array $property The property or properties to check.
+     * @return bool
+     */
+    public function has($property)
+    {
+        return $this->hasProperty($property);
+    }
+
+    /**
+     * Checks that a property is empty
+     *
+     * This is not working like the built in empty() of php. The method will
+     * return true for:
+     *
+     * - '' (empty string)
+     * - null
+     * - []
+     *
+     * but false on any other case.
+     *
+     * @param string $property The property to check.
+     * @return bool
+     */
+    public function isEmpty($property)
+    {
+        $value = $this->get($property);
+        if ($value === null
+            || (is_array($value) && empty($value)
+            || (is_string($value) && empty($value)))
```

Comment:
empty() on string values will fail on `'0'`. I think you want `strlen()` instead.